### PR TITLE
Login webform for retrive token

### PR DESCRIPTION
**Pull request by Ilia1**

### CODE REVIEW

**Comment by 0xf4b1** [comment]
Hi, cool idea and thank you for contributing! :)
The login could also be useful to provide region and language selection.
I just tried it but it does not yet work for me, I will look into this later again.

File: login/login.c [hunk 1]
```diff
@@ -28,14 +27,14 @@ static void checkUri (const char* uri)
...
-//printf("this not token!");	
-{
-	char* cmd = strcpy("mono token.exe ",token);
-	system(cmd);
-	gtk_main_quit();
-}
+	//printf("%s\n",token);
+	if (token[2] == '-' && token[35] == '-') 
+	{
+		char* cmd = strcpy("mono Token.exe ",token);
```
Comment (0xf4b1) [inline login/login.c:33]: I think the strcpy crashes for me

**Review by Ilia1 — COMMENTED** [review]
It seems like this should automatically redirect to the region. Please check

**Comment by 0xf4b1** [comment]
You are right, I think this is better now. But there is one more problem when extracting the token. I get redirected to `http://localhost/?ST=` which is 21 chars, but in `checkUri` you start with `int i = 23;` and it fails for me to get the token unless I change it. I think there is something different with another language?

**Comment by Ilia1** [comment]
Hmm, I was getting a link like http://localhost:0/?ST=

**Comment by Ilia1** [comment]
And maybe add a small script on the bash like
```
if [[ -f "token" ]]; then
        ./Bin/Hearthstone.x86_64
else
        ./login
        ./Bin/Hearthstone.x86_64
fi
```

**Comment by 0xf4b1** [comment]
> Hmm, I was getting a link like http://localhost:0/?ST=

Actually I get the same when I request the token manually, as I described it in the README. For some reason the `:0` is missing for me with the login app. But now it works for both :)
Maybe we should just make sure it also works for links not containing any `=` when clicking on the links to create an account or so

**Comment by 0xf4b1** [comment]
> And maybe add a small script on the bash like

I think we can assume there is no `token` when using the script because it is not really intended for updating the installation.
So lets just start the login app as the last step.

I prefer to leave the launch of the game to the user, but I think it would be cool to integrate a desktop entry to `~/.local/share/applications/` for starting the game

**Pull request merged**